File: app/src/main/java/com/akilimo/mobile/views/fragments/dialog/MaizePerformanceDialogFragment.java
package com.akilimo.mobile.views.fragments.dialog;


import android.app.Dialog;
import android.content.Context;
import android.content.DialogInterface;
import android.graphics.Color;
import android.graphics.drawable.ColorDrawable;
import android.os.Bundle;
import android.view.Window;
import android.widget.Button;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

import com.akilimo.mobile.R;
import com.akilimo.mobile.entities.MaizePerformance;
import com.akilimo.mobile.inherit.BaseDialogFragment;
import com.akilimo.mobile.interfaces.IMaizePerformanceDismissListener;
import com.akilimo.mobile.utils.Tools;

;

/**
 * A simple {@link androidx.fragment.app.Fragment} subclass.
 */
public class MaizePerformanceDialogFragment extends BaseDialogFragment {


    private static final String LOG_TAG = MaizePerformanceDialogFragment.class.getSimpleName();

    public static final String PERFORMANCE_DATA = "performance_data";
    public static final String ARG_ITEM_ID = "maize_performance_dialog_fragment";


    private boolean performanceConfirmed = false;

    private Dialog dialog;
    private Button btnClose;
    private Button btnConfirm;
    private TextView lblFragmentTitle;
    private TextView perfDescription;
    private ImageView performanceImage;
    private Button btnRemove;


    private MaizePerformance maizePerformance;

    private IMaizePerformanceDismissListener onDismissListener;

    public MaizePerformanceDialogFragment(Context context) {
        this.context = context;
    }


    @NonNull
    @Override
    public Dialog onCreateDialog(@Nullable Bundle savedInstanceState) {

        Bundle bundle = this.getArguments();
        if (bundle != null) {
            maizePerformance = bundle.getParcelable(PERFORMANCE_DATA);
        }
        dialog = new Dialog(context);

        dialog.getWindow().requestFeature(Window.FEATURE_NO_TITLE);
//        dialog.getWindow().setFlags(WindowManager.LayoutParams.FLAG_FULLSCREEN, WindowManager.LayoutParams.FLAG_LAYOUT_IN_SCREEN);
        dialog.setContentView(R.layout.fragment_maize_perf_dialog);

        dialog.setCancelable(true);
        dialog.getWindow().setBackgroundDrawable(new ColorDrawable(Color.TRANSPARENT));
        dialog.getWindow().getAttributes().windowAnimations = R.style.DialogSlideAnimation;


        btnClose = dialog.findViewById(R.id.close_button);
        btnConfirm = dialog.findViewById(R.id.confirm_button);
        btnRemove = dialog.findViewById(R.id.cancel_button);

        lblFragmentTitle = dialog.findViewById(R.id.lblFragmentTitle);
        perfDescription = dialog.findViewById(R.id.perfDescription);
        performanceImage = dialog.findViewById(R.id.rootYieldImage);

        if (maizePerformance != null) {
            String yieldAmountLabel = maizePerformance.getMaizePerformanceLabel();
            String yieldDesc = maizePerformance.getMaizePerformanceDesc();

            lblFragmentTitle.setText(yieldAmountLabel);
            perfDescription.setText(yieldDesc);

            Tools.displayImageOriginal(this.context, performanceImage, maizePerformance.getImageId());
        }


        btnClose.setOnClickListener(view -> {
            performanceConfirmed = false;
            dismiss();
        });

        btnRemove.setOnClickListener(view -> {
            performanceConfirmed = false;
            dismiss();
        });
        //save the data
        btnConfirm.setOnClickListener(v -> {
            performanceConfirmed = true;
            dismiss();
        });

        return dialog;
    }


    @Override
    public void onDismiss(@NonNull DialogInterface dialog) {
        super.onDismiss(dialog);
        if (onDismissListener != null) {
            onDismissListener.onDismiss(this.maizePerformance, performanceConfirmed);
        }
    }

    public void setOnDismissListener(IMaizePerformanceDismissListener dismissListener) {
        this.onDismissListener = dismissListener;
    }
}

File: app/src/main/java/com/akilimo/mobile/views/fragments/dialog/MaizePerformanceDialogFragment.kt
package com.akilimo.mobile.views.fragments.dialog

import android.app.Dialog
import android.content.DialogInterface
import android.graphics.Color
import android.os.Bundle
import android.view.View
import android.view.Window
import android.view.WindowManager
import androidx.core.graphics.drawable.toDrawable
import com.akilimo.mobile.R
import com.akilimo.mobile.databinding.FragmentMaizePerfDialogBinding
import com.akilimo.mobile.entities.CropPerformance
import com.akilimo.mobile.inherit.BaseDialogFragment
import com.akilimo.mobile.interfaces.ICropPerformanceListener
import com.akilimo.mobile.utils.Tools.displayImageOriginal


/**
 * A simple [androidx.fragment.app.Fragment] subclass.
 */
class MaizePerformanceDialogFragment : BaseDialogFragment() {
    private var performanceConfirmed = false


    private var cropPerformance: CropPerformance? = null

    private var onDismissListener: ICropPerformanceListener? = null
    private var _binding: FragmentMaizePerfDialogBinding? = null

    private val binding get() = _binding!!

    companion object {
        const val PERFORMANCE_DATA: String = "performance_data"
        const val ARG_ITEM_ID: String = "maize_performance_dialog_fragment"
    }

    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
        val bundle = this.arguments
        val context = requireContext()
        if (bundle != null) {
            cropPerformance = bundle.getParcelable(PERFORMANCE_DATA)
        }
        _binding = FragmentMaizePerfDialogBinding.inflate(layoutInflater)


        val dialog = Dialog(context)
        dialog.apply {
            window!!.requestFeature(Window.FEATURE_NO_TITLE)
            window!!.setFlags(
                WindowManager.LayoutParams.FLAG_FULLSCREEN,
                WindowManager.LayoutParams.FLAG_LAYOUT_IN_SCREEN
            )
            window!!.setBackgroundDrawable(Color.TRANSPARENT.toDrawable())
            window!!.attributes.windowAnimations = R.style.DialogSlideAnimation

            setContentView(binding.root)
        }

        if (cropPerformance != null) {
            val yieldAmountLabel = cropPerformance!!.maizePerformanceLabel
            val yieldDesc = cropPerformance!!.maizePerformanceDesc

            binding.lblFragmentTitle.text = yieldAmountLabel
            binding.perfDescription.text = yieldDesc

            displayImageOriginal(this.context, binding.rootYieldImage, cropPerformance!!.imageId)
        }


        binding.closeButton.setOnClickListener { view: View? ->
            performanceConfirmed = false
            dismiss()
        }

        binding.cancelButton.setOnClickListener { view: View? ->
            performanceConfirmed = false
            dismiss()
        }

        binding.confirmButton.setOnClickListener { v: View? ->
            performanceConfirmed = true
            dismiss()
        }

        return dialog
    }


    override fun onDismiss(dialog: DialogInterface) {
        super.onDismiss(dialog)
        if (onDismissListener != null) {
            onDismissListener!!.onDismiss(cropPerformance!!, performanceConfirmed)
        }
    }

    fun setOnDismissListener(dismissListener: ICropPerformanceListener?) {
        this.onDismissListener = dismissListener
    }


}
